src/main/java/org/kainos/ea/api/AuthService.java

```diff
@@ -0,0 +1,44 @@
+package org.kainos.ea.api;
+
+import com.password4j.Password;
+import org.kainos.ea.cli.LoginDetails;
+import org.kainos.ea.cli.LoginRequest;
+
+import org.kainos.ea.client.CouldNotGeneratePasswordHashException;
+import org.kainos.ea.client.CouldNotStoreNewJWTException;
+import org.kainos.ea.client.DatabaseConnectionFailedException;
+import org.kainos.ea.client.JWTCouldNotBeCreatedException;
+import org.kainos.ea.client.InvalidLoginAttemptException;
+import org.kainos.ea.client.CouldNotFindUserAccountException;
+
+import org.kainos.ea.db.AuthDAO;
+import org.kainos.ea.util.JwtGeneratorUtil;
+
+import java.security.NoSuchAlgorithmException;
+import java.sql.SQLException;
+import java.time.Clock;
+
+public class AuthService {
+
+  public String userLogin(LoginRequest loginRequest, AuthDAO authDAO) throws
+          SQLException, NoSuchAlgorithmException, DatabaseConnectionFailedException,
```

Comment:
`SQLException`, and `NoSuchAlgorithmException` can be removed

Reply:
Have been removed, thank you.